modules/docs/docgen/createTraversals.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * Creates `traversals.ts` from the source typescript.d.ts file for guards and the `find` function
+ */
+
+import fs from 'fs';
+import path from 'path';
+
+///** Create the guard fn
+const source = fs
+  .readFileSync(path.join(__dirname, '../../../node_modules/typescript/lib/typescript.d.ts'))
+  .toString();
+const lines = source.split('\n');
+
+const syntaxKinds = source.match(/export enum SyntaxKind \{([^}]+)\}/m);
+
+const pairs = syntaxKinds![1].match(/([a-z]+) = [\d]+/gi);
+
+const guardPairs: string[][] = [];
+
+pairs?.forEach(pair => {
+  const matches = pair.match(/([a-z]+) = [0-9]+/i);
+
+  if (matches) {
+    const syntaxKind = matches[1];
+
+    const line = lines.findIndex(l => l.includes(`kind: SyntaxKind.${syntaxKind};`));
```

Comment:
It would be a little easier to follow if `line` is named `lineIndex`. At a glance one might think it's an element in the `lines` array.

Reply:
Good call